x/gamm/keeper/estimate_swap.go

```diff
@@ -188,68 +154,11 @@ func (k Keeper) EstimateSwapExactAmountOut(
 		return sdk.Int{}, err
 	}
 
-	_, tokenIn, err := k.estimateSwapExactAmountOut(ctx, pool, tokenInDenom, tokenInMaxAmount, tokenOut, swapFee)
+	_, tokenIn, err := k.swapExactAmountOutNoTokenSend(ctx, pool, tokenInDenom, tokenInMaxAmount, tokenOut, swapFee)
 	if err != nil {
 		return tokenIn.Amount, err
 	}
 
-	err = k.estimateUpdatePoolForSwap(ctx, pool, tokenIn, tokenOut)
+	err = k.updatePoolForSwap(ctx, pool, tokenIn, tokenOut)
 	return tokenIn.Amount, err
 }
-
-func (k Keeper) estimateSwapExactAmountOut(
-	ctx sdk.Context,
-	pool types.PoolI,
-	tokenInDenom string,
-	tokenInMaxAmount sdk.Int,
-	tokenOut sdk.Coin,
-	swapFee sdk.Dec,
-) (updatedPool types.PoolI, tokenIn sdk.Coin, err error) {
-	if tokenInDenom == tokenOut.Denom {
-		return pool, sdk.Coin{}, errors.New("cannot trade same denomination in and out")
-	}
-
-	poolOutBal := pool.GetTotalPoolLiquidity(ctx).AmountOf(tokenOut.Denom)
-	if tokenOut.Amount.GTE(poolOutBal) {
-		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrTooManyTokensOut,
-			"can't get more tokens out than there are tokens in the pool")
-	}
-
-	tokenIn, err = pool.SwapInAmtGivenOut(ctx, sdk.Coins{tokenOut}, tokenInDenom, swapFee)
-	if err != nil {
-		return pool, sdk.Coin{}, err
-	}
-	tokenInAmount := tokenIn.Amount
-
-	if tokenInAmount.LTE(sdk.ZeroInt()) {
-		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrInvalidMathApprox, "token amount is zero or negative")
-	}
-
-	if tokenInAmount.GT(tokenInMaxAmount) {
-		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrLimitMaxAmount, "Swap requires %s, which is greater than the amount %s", tokenIn, tokenInMaxAmount)
-	}
-
-	return pool, tokenIn, nil
-}
-
-// estimateUpdatePoolForSwap updates the pool balance.
-// This method does not include emitting hooks or sending the actual asset from
-// the sender to the pool or vice versa, as this is designed to be used by query methods.
-func (k Keeper) estimateUpdatePoolForSwap(
```

Comment:
changed to method `updatePoolForSwap` in swap.go, as this should also not be an estimating method